core/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -149,13 +155,19 @@ public Reason getReason() {
     }
 
     /**
-     * The timestamp in milliseconds since epoch. Note, we use timestamp here since
-     * we want to make sure its preserved across node serializations. Extra care need
-     * to be made if its used to calculate diff (handle negative values) in case of
-     * time drift.
+     * The timestamp in milliseconds since epoch (based on System.currentTimeMillis()).
+     * Note, we use timestamp here since we want to make sure its preserved across node serializations.
      */
     public long getTimestampInMillis() {
-        return this.timestamp;
+        return this.timestampMillis;
+    }
+
+    /**
+     * The timestamp in nanoseconds (based on System.nanoTime()). Used to calculate the delay for delayed shard allocation.
+     * ONLY EXPOSED FOR TESTS!
+     */
+    public long getDelayCalculationTimestampInNanos() {
```

Comment:
Oh, got confused , which is the point :) getDelayCalculationTimestampInNanos -> getUnassignedTimeInNanos